homeassistant/components/anova_sous_vide/config_flow.py

```diff
@@ -0,0 +1,34 @@
+"""Config flow for Anova."""
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+
+class AnovaConfligFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Sets up a config flow for Anova."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        if user_input is not None:
+            entries = self._async_current_entries()
+            if any(x.data["device_id"] == user_input["device_id"] for x in entries):
+                return self.async_abort(reason="already_configured")
+
+            await self.async_set_unique_id(user_input["device_id"])
```

Comment:
https://developers.home-assistant.io/docs/config_entries_config_flow_handler?_highlight=unique&_highlight=id#example-acceptable-sources-for-a-unique-id

The unique id should come from the device

Reply:
the device_id does come from the device, the user just has to get it as we need it to be able to communicate with the device at all. The id is unique.

There are no other unique identifiers from the api call and one username can have multiple devices, so that wouldn't work either.

What would you recommend?

Reply:
https://github.com/Lash-L/anova_wifi/blob/3a351a416036f431de291d92e2169b91008ddb42/src/anova_wifi/parser.py#L50

oh my. no authentication at all?

Reply:
Let's call this `device_unique_id` so its clear that its globally unique

Reply:
> no authentication at all

yeah, their api is completely unprotected for getting data. 

trying to set anything is protected with a client secret, but I have yet to determine the best way to handle getting that/ making sure it is simple to get/ I may be able to figure out how to do it locally.

And for the sake of making sure this pr was small, the integration is just data pulling only.